### DISCUSSION TITLE
Questions Regarding FATES in CTSM5.3

### DISCUSSION THREAD

**Discussion by changemode** [NGEET/fates-tutorial#64] [Q&A]
Hi, Developers,

Thank you for providing the new FATES tutorial. Following the instructions, I successfully ran the example cases within CTSM5.3.009 (sci.1.78.3_api.36.1.0) on my HPC. As a new user of the FATES model, I have some questions that I hope the community can help me understand:
1.  Does FATES in CTSM5.3 run based on the SP (16 PFTs) or the BGC module (78 PFTs)? I created a case with **I1PtClm60Fates**, and I noticed that the tutorial’s BCI case uses 16-PFT surface data in ELM.
2. Regarding model initialization, the tutorial used the “cold-start” with inventory initialization. However, according to CTSM documentation, SP and BGC simulation typically require spin-up of at least 50 and several hundred years, respectively, to achieve equilibrium. My question is: Is a spinup necessary when using inventory initialization in CTSM-FATES? My current idea is to use the“cold-start” with inventory initialization for spinup, generate a restart file and then proceed with the simulation. Is this approach reasonable? If so, how many years of spinup are generally recommended for FATES?
3. FATES allows selecting PFT indices through the parameter netCDF file to define the vegetation composition for my study site. For example, if I use the Python script with --pft-indices [13,13] to specify the PFT, my study site will be restricted to using this specific PFT (cool_c3_grass), and no other PFTs will be allowed to grow. I can further modify the different parameters for this PFT to represent two distinct vegetations within the same PFT (cool_c3_grass). Is this understanding correct?

I appreciate your time and look forward to your insights.

Best regards,
zihang

**Comment by glemieux** [comment]
Hello @changemode, thanks for reaching out.

1. FATES does not yet have the capability to run with 78 pfts.  There is a ctsm issue that needs to be addressed to facilitate this which you can track here: https://github.com/ESCOMP/CTSM/issues/1785.  There is also some discussion about the issue that you can find here: https://github.com/ESCOMP/CTSM/issues/2810#issuecomment-2397521191
2. My understanding is that spin-up is not necessary when using inventory initialization, but let me check on that.
3. Yes this understanding is correct.

**Reply by changemode** [reply]
Hi @glemieux, thank you so much for your detailed and prompt response! Additionally, I noticed that write(fates_log(), *) in the source code, which seems to be used for outputting log information related to the model's execution. However, I couldn't find the Fates log files in the run directory or within the lnd.log file. Could you advise where these logs are stored? Thank you for your assistance!

**Reply by glemieux** [reply]
Good question! The `fates_log` output gets placed in the lnd.log file along with the host land model log outputs.  There is no specific separate fates log file.

**Reply by changemode** [reply]
thank you @glemieux, I have some questions while running the Fates model within CTSM5.3:
1. I am running a single-point simulation in the Qinghai-Tibet Plateau using inventory initialization with only one PFT (index 12, arctic_c3_grass) defined in the FATES parameters. I've observed that during the initial 7-year simulation, FATES_LAI shows a significant spinup process with gradual increase, but I rerun 7-year simulation with the restart file that from inventory initialization, FATES_LAI exhibits more stable and reasonable. I don't know if my css and pss file is not suitable and get the restart file through the “cold-start” with inventory initialization is resonable. This is my css and pss file and two experiments FATES_LAI results.

      pss:
       time patch trk age area
       0000 <patch_0001> 0 0.0000000000000000 1
    
      css:
       "time" "patch" "dbh" "height" "pft" "plant"
      0 "<patch_0001>" -3 0.02 0 500
![1](https://github.com/user-attachments/assets/612e9537-a007-4516-80a2-869f41fdf838)

2. Regarding the css file for my study site, I wonder if the PFT indices in the css file should be zero-based, counting up to the number of PFTs. Initially, I set it to match the FATES parameter index (12) but received an error. After checking FatesInventoryInitMod.F90, I found the condition 'c_pft > numpft' on line 877, so I modified the PFT in the css file to 0. Could you confirm if this is the correct approach?

3. The LAI drops to zero in winter. Does this indicate vegetation mortality during the winter season? The winter LAI values seem a bit too low. Could you provide some suggestions on how to address this issue of zero winter LAI in the simulation results, either through modifying FATES parameter files or adjusting parameterization schemes in the code?

**Reply by rgknox** [reply]
Hello @changemode ,  thanks for starting this discussion.
Could you provide us with some more details on how you set up these runs?   I think the model output you are seeing is explainable, but with some more information, we can be sure the runs are setup the way you want them.

1) How did you modify the parameter file to use just one PFT? The best way to select a single PFT from the parameter file and create a new file is to use the [tools/FatesPFTIndexSwapper.py](https://github.com/NGEET/fates/blob/main/tools/FatesPFTIndexSwapper.py) python script.  Could you share with us the resulting parameter file you generated, and if need be, feel free to remove any proprietary parameters if that is of concern.

2) We wanted to make sure that the "cold start' inventory is being started correctly for your first run.  When the inventory initialization is used, it will write some lines of text to the output logs about the basal area before and after fusion. See [here](https://github.com/NGEET/fates/blob/main/main/FatesInventoryInitMod.F90#L415-L419). Can you look in your run logs to see if that message appears (could be either the cesm.log or the lnd.log).

3) I wasn't sure about the timing of the restart.  What is the timestamp of the restart file that is used to start the second simulation.  Specifically, is that file from the very beginning or the very end of the first simulation. And, what type of restart was this, is it a "continue_run" restart (which would imply using a restart file at the end of your first run), or is it one where you specify "finitdat=" in your namelist (which could point to any restart file you choose).

If you could provide us any logs, and your lnd_in files, that would be very helpful too.

**Reply by changemode** [reply]
Thank you for your detailed response @rgknox. I apologize for not providing sufficient details about the model setup initially.
I created the case using:
`./create_newcase --case Madoi_1 --res CLM_USRDAT --compset I1PtClm60Fates --compiler intel --run-unsupported`
I conducted three FATES simulation runs (covering the period from 2012/1/1 to 2018/12/31), all using a single-PFT (index 12, arctic_c3_grass) parameter file:
The first run (shown in blue line) was a cold start with inventory initialization and default [single-PFT](https://github.com/user-attachments/files/18003521/default_only_12pft.txt) parameter file. This is [lnd_in](https://github.com/user-attachments/files/18005163/lnd_in.txt).

The second run (shown in yellow line) utilized a restart file from my first cold-start simulation by setting "finitdat=2019-01-01.r.nc" in the namelist, the restart file was generated at the end of my first cold-start simulation (2012.1.1-2018.12.31)

The third run (shown in red line) was based on the second run with only modified [some parameters](https://github.com/user-attachments/files/18003522/modify_only_12pft.txt) in single-PFT parameter file to better match my study site.

The LAI shows:
![image](https://github.com/user-attachments/assets/a934a272-6d22-4b3f-9a22-2ad94c4c531f)
1. Yes, I used the tools/FatesPFTIndexSwapper.py Python script and I attachted parameter files.
2. I am sure that the inventory initialization is used through that you mentioned in [lnd.log](https://github.com/user-attachments/files/18003761/lnd.log.4979634.241128-101020.txt).
 Additionally, I noticed the "plant" is 500/m2 in my css is too large, so the number of plants shows:
 
![image](https://github.com/user-attachments/assets/2e72783a-d96e-413b-a4b9-0fe1253f93a0)

 
 3. I used "finitdat=*2019-01-01.r*.nc" in my namelist , which is the restart file that was generated at the end of my first cold-start simulation (2012.1.1-2018.12.31).
 
 My question is:
 Q1. I am curious about that regarding the PFT indices in css file and FATES parameter index.  I mentioned a similar question in the second issue of my previous post.
 
 Q2. The LAI drops to zero in winter while the number of plant is about 1, which means complete leaf senescence? The winter LAI values and all years number of plant seem a bit too low. Could you provide some suggestions on how to address this issue of zero winter LAI and number of plant in the simulation results, either through modifying FATES parameter files or adjusting parameterization schemes in the code?
 
 Q3. The simulated LAI and number of plants show opposite trends in first run (cold start with inventory initialization). Due to an unrealistically high initial plant density setting (500/m2) in the css file, as the simulation progresses, we observe a decrease in plant numbers but an increase in LAI until both reach equilibrium. Since I have limited knowledge of vegetation dynamics, shouldn't the more number of plant there be, the greater the LAI?
 
 Q4. I used the“cold-start” with inventory initialization for spinup, generate a restart file and then proceed with the simulation. Is this approach reasonable? If so, how many years of spinup are generally recommended for FATES?
 
 I really appreciate your response, just hoping to get a bit more clarification on this point if possible. Thank you again!
